life_expectancy/load_save_data.py

```diff
@@ -1,23 +1,77 @@
 """This model loads and saves the data"""
-from pathlib import Path
+#pylint: disable=R0903
 
+from pathlib import Path
+from typing import Any, Protocol
+import zipfile
 import pandas as pd
 
 
-def load_data(
-        input_file_path: Path
-    ) -> pd.DataFrame:
-    """
-    Load eu_life_expectancy_raw.tsv data from the data folder
-    Args:
-        input_file_path (Path): Path for the file to be loaded
-    Returns:
-        life_exp_raw_data (Pandas DataFrame): DataFrame with data to be cleaned
-    """
-    with open(input_file_path, "r", encoding="UTF-8") as file:
-        life_exp_raw_data = pd.read_csv(file, sep="\t")
+class FileReaderStrategy(Protocol):
+    """Reads the file"""
+
+    def __call__(self, file:Any) -> pd.DataFrame:
```

Comment:
**Suggestion**: This is just a comment, and not a change request. I'm not a fan of using the `__call__` method like his. I assume you are using it so that your classes work as functions... but in this case, why not use functions directly? Take Arjan's example as inspiration https://youtu.be/n2b_Cxh20Fw?t=1365

Reply:
I have changed the logic to Protocols without the using of __call__ dunder methods. Since I wanted also to add in that class the cleaning behaviour that JSON/TSV should take I have used Protocol (with two subclasses JSONFileHandler and  TSVFileHandler), with two methods, one named load_file and the other one clean_data. Thanks :)